### PR TITLE
KRKPD-1083: Refactor and expand testing

### PR DESCRIPTION
**Pull request by klawil**
## Type of change
- [x] Refactoring (no functional changes, no api changes)

## Description of change
See [here](https://kargo1.atlassian.net/browse/KRKPD-1083) for easily readable map of old and new tests
- Refactored the tests for the Kargo bid adapter. The tests were refactored to expand coverage of edge cases and invalid data being provided.
- Updates from 29 defined tests to 81 for more detailed unit tests.

### CODE REVIEW

**Comment by nickllerandi** [comment]
Going through this one function/section at a time:
- [x] isBidRequestValid
- [x] buildRequests (revisit after addressing device h/w test)
- [x] interpretResponse
- [x] getUserSyncs
- [x] supportedMediaTypes
- [x] onTimeout

File: test/spec/modules/kargoBidAdapter_spec.js [hunk 1]
```diff
@@ -1,461 +1,1481 @@
...
+
+      // User
+      expect(payload.user).to.be.an('object');
+      expect(payload.user.crbIDs).to.deep.equal({});
+      expect(payload.user.data).to.deep.equal([]);
+      expect(payload.user.sharedIDEids).to.deep.equal(defaultBidParams.userIdAsEids);
+
+      // General keys
+      expect(payload.aid).to.equal('randomAuctionId');
+      expect(payload.device).to.deep.equal({ size: [ 3440, 1440 ] });
```
Comment (nickllerandi) [inline test/spec/modules/kargoBidAdapter_spec.js]: This one is failing on my end, definitely because the adapter is referencing `window` to get this information. Maybe  let's just set a device h/w using that `getPayloadFromTestBids` function you set up.
https://github.com/KargoGlobal/Prebid.js/blob/master/modules/kargoBidAdapter.js#L91-L94

Comment (klawil) [reply test/spec/modules/kargoBidAdapter_spec.js]: I opted to just pull the values from the window object rather than set them on the window object to avoid mutating the global or having to set up a new sandbox to override the window global without affecting other adapter tests, thoughts on that approach?

File: test/spec/modules/kargoBidAdapter_spec.js [hunk 2]
```diff
@@ -1050,148 +1785,205 @@ describe('kargo adapter tests', function () {
...
-    beforeEach(function () {
-      triggerPixelStub = sinon.stub(utils, 'triggerPixel');
+    it('does not call triggerPixel if timeout data is not provided', function() {
+      spec.onTimeout(null);
+      expect(utils.triggerPixel.callCount).to.equal(0);
     });
 
-    afterEach(function () {
-      utils.triggerPixel.restore();
+    it('calls triggerPixel if any timeout data is provided', function() {
```
Comment (nickllerandi) [inline test/spec/modules/kargoBidAdapter_spec.js:1971]: Prob don't need two of these, right?
https://github.com/KargoGlobal/Prebid.js/pull/31/files#diff-2fe18725d9d2620649c2894853c6a5f75ca201cad0b65ebdfa37ac37d48e676eR1976

Comment (klawil) [reply test/spec/modules/kargoBidAdapter_spec.js:1971]: Good callout, I will remove the first test

**Review by fionasequeira — APPROVED** [review]

**Review by njflynn — APPROVED** [review]
